Upgrading note for legacy collector sidecar removal (#16343)
Graylog2/graylog2-server: UPGRADING.md

## Breaking Changes

> - Removing support for legacy "Collector Sidecars".
> 
>   Graylog 3.0 introduced "Graylog Sidecars" as an replacement for the old Collector Sidecars (version 0.1.x).
> 
>   With Graylog 5.2 support for the legacy Collector Sidecars will be finally removed.
> 
>   Please refer to the migration guide [Upgrading from the Collector Sidecar](https://archivedocs.graylog.org/en/3.3/pages/sidecar.html#upgrading-from-the-collector-sidecar)
>   if you are still using the old Sidecars before upgrading Graylog to 5.2.
> 

Reviewer: Let's use a header and section for the notes.

```suggestion
### Removed support for legacy "Collector Sidecars"

Graylog 3.0 introduced "Graylog Sidecars" as a replacement for the old Collector Sidecars (version 0.1.x).

With Graylog 5.2, support for the legacy Collector Sidecars is finally removed.

Please refer to the migration guide [Upgrading from the Collector Sidecar](https://archivedocs.graylog.org/en/3.3/pages/sidecar.html#upgrading-from-the-collector-sidecar) if you are still using the old Sidecars before upgrading Graylog to 5.2.

```
